### DISCUSSION TITLE
WebSocket connection failed

### DISCUSSION THREAD

**Discussion by emretuncer** [appbaseio/reactivesearch#1538] [General]
**Affected Projects**
Vue.JS
**Library Version:** 1.7.5
**Describe the bug**
`WebSocket connection to 'wss://localhost:9200/questions' failed: Error in connection establishment: net::ERR_SSL_PROTOCOL_ERROR`

-> appbase-js.es.js?bed6:361

**To Reproduce**
```
            <ReactiveBase
                    app="questions"
                    url="http://localhost:9200/">
```
**Expected behavior**

**Desktop (please complete the following information):**
All Browsers

**Additional context**
I tried too many thing to solve that bug but i didn't find a solution. (ReactiveList Component)

> I was able to get it to work. The missing piece was setting the correct access control for my local environment. For example:
> 
> ```
> http.cors.enabled: true
> http.cors.allow-credentials: true
> http.cors.allow-origin: /https?:\/\/(localhost)?(127.0.0.1)?(:[0-9]+)?/
> http.cors.allow-headers: X-Requested-With, X-Auth-Token, Content-Type, Content-Length, Authorization, Access-Control-Allow-Headers, Accept
> ```

-----------

> **TLDR:**
> 
> - [Proxy server][2]
> - Using the proxy server in [client app][3] with reactivesearch

> The connection error related to websockets you see here isn't causing the issue. It's used for streaming which works on appbase.io. This has been fixed in the `2.2.0` release. Hope this helps :)

It is said to be resolved in the react version but there is no solution for vue.

  [2]: https://github.com/appbaseio-apps/reactivesearch-proxy-server
  [3]: https://github.com/appbaseio-apps/reactivesearch-proxy-client

**Comment by bietkul** [comment]
Vue library doesn't support web sockets.
